sklearn/datasets/_samples_generator.py

```diff
@@ -1510,6 +1510,16 @@ def make_spd_matrix(n_dim, *, random_state=None):
     return X
 
 
+@validate_params(
+    {
+        "dim": [Interval(Integral, 1, None, closed="left")],
+        "alpha": [Interval(Real, 0, 1, closed="both")],
+        "norm_diag": ["boolean"],
+        "smallest_coef": [Interval(Real, 0, 1, closed="both")],
+        "largest_coef": [Interval(Real, 0, 1, closed="both")],
```

Comment:
I don't think `smallest/largest_coef` are constrained.
```suggestion
        "smallest_coef": [Interval(Real, None, None, closed="neither")],
        "largest_coef": [Interval(Real, None, None, closed="neither")],
```

Reply:
Uhm the documentation state it should be 0 and 1.